source/acquaman/SGM/SGMLineScanConfiguration.h

```diff
@@ -0,0 +1,37 @@
+#ifndef SGMLINESCANCONFIGURATION_H
+#define SGMLINESCANCONFIGURATION_H
+
+#include "acquaman/AMGenericContinuousScanConfiguration.h"
+
+/// This class implements SGM spatial line scanning.
+class SGMLineScanConfiguration : public AMGenericContinuousScanConfiguration
+{
+	Q_OBJECT
+
```

Comment:
I presume no Q_PROPERTY()s are needed because the superclass configuration is handling all of the mundane details for selected detectors and controls and whatnot.